### PR TITLE
[Inductor] Support custom op in JIT with cpp wrapper

### PR DESCRIPTION
**Pull request by desertfire**
Summary:  To call custom ops in an ABI-compatible way requires doing boxed call with varargs across C shim. In the JIT mode, we can get around it by calling into Python.  https://gist.github.com/desertfire/be2a65b0a9b47780bb716b53ac2cd2b3 is an example of generated code.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #122554

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @chauhang

Differential Revision: [D55326556](https://our.internmc.facebook.com/intern/diff/D55326556)

### CODE REVIEW

File: torch/_inductor/codegen/cpp_wrapper_cpu.py [hunk 1]
```diff
@@ -999,6 +1000,14 @@ def generate_before_suffix(self, result):
             if V.graph.aot_mode:
                 result.writeline("} // AOTInductorModel::run_impl")
             else:
+                if self.custom_op_wrapper_loaded:
+                    with result.indent():
+                        result.splice(
+                            """
+                            Py_DECREF(codecache_module);
```
Comment (jansel) [inline torch/_inductor/codegen/cpp_wrapper_cpu.py:1007]: Should we add a small RAII wrapper for all the decrefs?

**Review by jansel — CHANGES_REQUESTED** [review]

**Comment by desertfire** [comment]
Update with a `RAIIPyObject` class. Updated the gist link in the comment to https://gist.github.com/desertfire/be2a65b0a9b47780bb716b53ac2cd2b3

**Review by jansel — APPROVED** [review]

**Comment by desertfire** [comment]
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D55326556).

**Review by chenyang78 — APPROVED** [review]
LGTM. Thanks!

**Comment by jeffdaily** [comment]
This did break ROCm inductor CI.  I opened a DISABLED issue to skip.  https://github.com/pytorch/pytorch/issues/122811

**Pull request closed without merging**